engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/ExecuteJob.scala

```diff
@@ -35,6 +35,7 @@ class ExecuteJob(
           this.getClass,
           () => {
             try {
+              ctx.state.executionHooks.run()
```

Comment:
But here you only assume to posses a read compilation lock whereas the one hook we are now adding requires a write one.

Reply:
You're right, for some reason, I decided that the execution requires a write compilation lock, and removed one from the hook body.
I suppose, hooks should be executed outside of the main runtime lock, and have their own locking logic.